Sources/System/FilePath/FilePathString.swift

```diff
@@ -0,0 +1,251 @@
+  }
+}
+
+
+
+// @available(macOS 10.16, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
+extension FilePath: ExpressibleByStringLiteral {
+  /// Creates a file path from a string literal.
+  ///
+  /// - Parameter stringLiteral: A string literal
+  ///   whose Unicode encoded contents to use as the contents of the path.
+  public init(stringLiteral: String) {
+    self.init(stringLiteral)
+  }
+
+  /// Creates a file path from a string.
+  ///
+  /// - Parameter string: A string
+  ///   whose Unicode encoded contents to use as the contents of the path.
+  public init(_ string: String) {
+    self.init(SystemString(string))
+  }
+}
+
+// @available(macOS 10.16, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
+extension String {
+  /// Creates a string by interpreting the file path's content as UTF-8 on Unix
+  /// and UTF-16 on Windows.
+  ///
+  /// - Parameter path: The file path to be interpreted as `PlatformUnicodeEncoding`.
+  ///
+  /// If the content of the file path isn't a well-formed Unicode string,
+  /// this initializer replaces invalid bytes them with U+FFFD.
+  /// This means that, depending on the semantics of the specific file system,
+  /// conversion to a string and back to a path
+  /// might result in a value that's different from the original path.
+  public init(decoding path: FilePath) {
+    self = path.withPlatformString { String(platformString: $0) }
+  }
+
+  /// Creates a string from a file path, validating its contents as UTF-8 on Unix
+  /// and UTF-16 on Windows.
+  ///
+  /// - Parameter path: The file path to be interpreted as `PlatformUnicodeEncoding`.
+  ///
+  /// If the contents of the file path isn't a well-formed Unicode string,
+  /// this initializer returns `nil`.
+  public init?(validating path: FilePath) {
+    guard let str = path.withPlatformString(
+      String.init(validatingPlatformString:)
+    ) else {
+      return nil
+    }
+    self = str
+  }
+}
+
+
+// @available(macOS 10.16, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
+extension FilePath: CustomStringConvertible, CustomDebugStringConvertible {
+  /// A textual representation of the file path.
+  ///
+  /// If the content of the path isn't a well-formed Unicode string,
+  /// this replaces invalid bytes them with U+FFFD. See `String.init(decoding:)`
+  @inline(never)
+  public var description: String { String(decoding: self) }
+
+  /// A textual representation of the file path, suitable for debugging.
+  ///
+  /// If the content of the path isn't a well-formed Unicode string,
+  /// this replaces invalid bytes them with U+FFFD. See `String.init(decoding:)`
+  public var debugDescription: String { description.debugDescription }
+}
+
+// @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+extension String {
+  /// Creates a string by interpreting the path component's content as UTF-8 on Unix
+  /// and UTF-16 on Windows.
+  ///
+  /// - Parameter path: The path component to be interpreted as `PlatformUnicodeEncoding`.
```

Comment:
The initializer methods in this extension appear to be documented incorrectly. Please correct me if I am wrong, but I believe IDEs would expect `- Parameter path:` to describe a parameter that is not named in any of these signatures. Should these documentation comments instead read `- Parameter component`?

Also, below is another instance of "invalid bytes them".

Reply:
J